pandas/core/series.py

```diff
@@ -3238,7 +3240,8 @@ def _try_cast(arr, take_fast_path):
         else:
             subarr = maybe_convert_platform(data)
 
-        subarr = maybe_cast_to_datetime(subarr, dtype)
+        if try_cast_to_datetime:
```

Comment:
why is this changed?

Reply:
sorry about that, i was trying out something similar to infer_datetimelike_array in that function before you told me to try in inference.pyx. I'll change it back to the original version.